### PR TITLE
For #12520: integrate changes from shotgunsoftware

### PR DESCRIPTION
**Pull request by ksallee**
The biggest change that kind of clashed with what we have changed is the addition of the `published_file_filters` setting. Since we added some `get_published_files_for_items_data` in our manager/hook setup, I mimicked the changes that were done to `get_published_files_for_items` (i.e. adding a filters param to the hook method and setting it to the user setting when called)

Other than that, all the changes come from shotgunsoftware. I tested as much as I could and it does not seem to break.
We have a base `get_published_files.py` hook in our `tk-config-gplstudio` that also needs a slight update to add the `filters` param.

### CODE REVIEW

File: hooks/get_published_files.py [hunk 1]
```diff
@@ -92,7 +94,7 @@ def get_published_files_for_items(self, items, data_retriever=None):
                     # If there's no data for this field, we still need to make sure
                     # there's an empty list for it to build the filters.
                     sg_data_by_field[field] = []
-        filters = []
+        filters = filters or []
```
Comment (gplsteph) [inline hooks/get_published_files.py]: Hmm, aren't we modifying the passed `filters`?

**Review by gplsteph — CHANGES_REQUESTED** [review]
Not sure I commented on what was needed, but I spotted a potential bug.

File: hooks/get_published_files.py (hunk 1 above)
Comment (ksallee) [reply hooks/get_published_files.py]: fixed

File: hooks/get_published_files.py [hunk 2]
```diff
@@ -42,13 +43,14 @@ def get_published_files_for_items_data(self, items_data, fields):
 
         :param items_data: A list of dictionaries as returned by the scene scanner.
         :param fields: A list of fields to query from SG.
+        :param filters: A list of filters to use when querying SG.
```
Comment (gplsteph) [inline hooks/get_published_files.py]:
```suggestion
        :param filters: An optional list of filters to use when querying SG.
```

File: hooks/get_published_files.py [hunk 3]
```diff
@@ -71,6 +73,7 @@ def get_published_files_for_items(self, items, data_retriever=None):
         :param items: a list of :class`FileItem` we want to get published files for.
         :param data_retriever: If provided, the api request will be async. The default value
             will execute the api request synchronously.
+        :param filters: A list of filters to use when querying SG.
```
Comment (gplsteph) [inline hooks/get_published_files.py]:
```suggestion
        :param filters: An optional list of filters to use when querying SG.
```

**Review by gplsteph — APPROVED** [review]
Looks good, just some suggestions for some params

**Pull request merged**